build-tools/scripts/Packaging.mk

```diff
@@ -28,6 +29,7 @@ create-vsix:
 		$(if $(PACKAGE_HEAD_BRANCH),/p:XAVersionBranch="$(PACKAGE_HEAD_BRANCH)") \
 		$(if $(PACKAGE_VERSION_REV),/p:XAVersionCommitCount="$(PACKAGE_VERSION_REV)") \
 		$(if $(COMMIT),/p:XAVersionHash="$(COMMIT)") \
+		$(if $(USE_COMMERCIAL_INSTALLER_NAME),/p:UseCommercialInstallerName="$(USE_COMMERCIAL_INSTALLER_NAME)") \
```

Comment:
Just a question... we already are doing this in this file a lot, but...

Do we need to create `$(USE_COMMERCIAL_INSTALLER_NAME)`?

Can you just do:
```
make foo USECOMMERCIALINSTALLERNAME=True
# or
make foo UseCommercialInstallerName=True
```
All env vars double as MSBuild properties, so we could stop doing this pattern?

Reply:
Oh this is interesting, I remember you mentioning this previously but it didn't click. I'll try to get this working without needing to explicitly pass the property to the msbuild invocation.

Reply:
@jonathanpeppers After some investigation, it seems that `make` doesn't provide a great way of exporting environment variables to child processes... I believe this could be achieved by potentially exporting variables in a shell invocation similar to `$(shell export foo=bar msbuild *.csproj)` but this quickly removes the benefits gained from trying to use variables here rather than adding properties to the msbuild invocation.

Reply:
So an example:

`foo.targets`:
```xml
<Project>
  <Target Name="Build">
    <Error Text="Bar: $(Bar)" />
  </Target>
</Project>
```

`Makefile`:
```
all:
	msbuild foo.targets
```

`make Bar=baz` returns:
```
foo.targets(3,5): error : Bar: baz
```
So it looks to me the env vars are working? Or do I need to run another process from _inside_ MSBuild?

Reply:
Seems to also work if I go one deeper:
```xml
<Project>
  <Target Name="Build">
    <Exec Command="sh foo.sh" />
  </Target>
</Project>
```
```bash
# foo.sh
echo "bar=$bar"
```

Reply:
Ahh thanks for the more detailed summary, I was trying to set this variable in the makefile itself, not when calling make. The following command does work as you suggested:

```
make create-installers UseCommercialInstallerName=true PkgLicenseSrcEn=path/to/commercial-license.txt
```

The problem I have with this approach is that it requires conditional make invocations in `build-tools\automation\build.groovy` (or in our Azure pipeline). For example, OSS builds would need to call
```
make create-installers
```
while commercial builds would call
```
make create-installers UseCommercialInstallerName=true PkgLicenseSrcEn=path/to/commercial-license.txt
```

The current approach allows you to only call `make create-installers` and the right naming/licenses will be used automatically in the installers for OSS builds or proprietary builds depending on whether or not the proprietary bits exist.